src/main/java/com/google/netpcapanalysis/models/MockPCAP.java

```diff
@@ -0,0 +1,51 @@
+package com.google.netpcapanalysis.models;
+
+import com.google.netpcapanalysis.interfaces.models.PCAP;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class MockPCAP implements PCAP {
+
+  private String id;
+  private List<InetAddress> ips;
+
+  public MockPCAP() {
+    this.id = "sampleId";
+    this.ips = new ArrayList<>();
+    for (int i = 0; i < 1000; i++) {
+      ips.add(genIP());
+    }
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  @Override
+  public List<InetAddress> getIPs() {
+    return ips;
+  }
+
+  public static InetAddress genIP() {
+    InetAddress ip = null;
+    Random r = new Random();
+    while (ip == null || ip.isLoopbackAddress() ||
+        ip.isAnyLocalAddress() ||
+        ip.isLinkLocalAddress() ||
+        ip.isSiteLocalAddress()) {
+      try {
+        ip = InetAddress.getByName(
+            r.nextInt(256) + "." +
+                r.nextInt(256) + "." +
+                r.nextInt(256) + "." +
+                r.nextInt(256)
+        );
```

Comment:
Did you do the rough math or a statistical computation on roughly how many iterations it takes before you get the IP you need. Seems a bit overkill for just mock data. You could as well run this offline in a separate script and then copy a set of valid IPs to a text file.

Reply:
I just copied it from the benchmark script. The time is pretty negligible. I used it to generate IPs for the benchmark script and 400K addresses didn't make the script much slower.